homeassistant/components/google_tasks/api.py

```diff
@@ -78,4 +96,43 @@ async def patch(
             task=task_id,
             body=task,
         )
-        await self._hass.async_add_executor_job(cmd.execute)
+        await self._execute(cmd)
+
+    async def delete(
+        self,
+        task_list_id: str,
+        task_ids: list[str],
+    ) -> None:
+        """Delete a task resources."""
+        service = await self._get_service()
+        batch: BatchHttpRequest = service.new_batch_http_request()
+
+        def response_handler(_, response, exception: HttpError) -> None:
+            if exception is not None:
+                raise GoogleTasksApiError(
+                    f"Google Tasks API responded with error ({exception.status_code})"
+                ) from exception
+            data = json.loads(response)
+            _raise_if_error(data)
```

Comment:
Does only the delete API return errors with an `error` key in the returned dict, or could this also be applied to some of the other api calls here like list or insert?

Reply:
The others apply this too, for example in `_execute` below on line 137. Delete is just different because it is a batch API, but it is sharing that part.